### PR TITLE
Added get/set for IsMaximized in Windows/Linux/BasicGameForm

### PR DESCRIPTION
**Pull request by guusw**
To allow storing in save files if the user has the window maximized and restore that setting. Also just a general handy method to have.

### CODE REVIEW

File: osu.Framework.Desktop/OS/Linux/LinuxGameForm.cs [hunk 1]
```diff
@@ -11,8 +11,14 @@ namespace osu.Framework.Desktop.OS.Linux
...
     public class LinuxGameForm : BasicGameForm
     {
-        public override bool IsMinimized => ClientSize.Width != 0 || ClientSize.Height == 0;
-        
+        public override bool IsMinimized => ClientSize.Width != 0 || ClientSize.Height == 0;
+
+        public override bool IsMaximized
+        {
+            get { return this.WindowState == FormWindowState.Maximized; }
+            set { this.WindowState = value ? (FormWindowState.Maximized) : (FormWindowState.Normal); }
```
Comment (peppy) [inline osu.Framework.Desktop/OS/Linux/LinuxGameForm.cs]: pointless brackets around enums

File: osu.Framework.Desktop/OS/Linux/LinuxGameForm.cs [hunk 2]
```diff
@@ -11,8 +11,14 @@ namespace osu.Framework.Desktop.OS.Linux
 {
     public class LinuxGameForm : BasicGameForm
     {
-        public override bool IsMinimized => ClientSize.Width != 0 || ClientSize.Height == 0;
-        
+        public override bool IsMinimized => ClientSize.Width != 0 || ClientSize.Height == 0;
+
+        public override bool IsMaximized
```
Comment (peppy) [inline osu.Framework.Desktop/OS/Linux/LinuxGameForm.cs]: why not implement this in the BasicGameForm since it's shared between both forms?

**Comment by guusw** [comment]
Makes sense to move it to BasicGameForm, then it also makes sense to move IsMinimized since that is also the same in both implementations.

**Comment by peppy** [comment]
being able to maximize the form and check maximized is probably not a thing we need, to be honest. "maximized" games should be borderless.

**Comment by guusw** [comment]
Allright, I guess this isn't needed then if borderless is used everywhere instead.

**Pull request closed without merging**